tests/main.cpp

```diff
@@ -35,11 +54,11 @@ int main(int argc, char* argv[]) {
   // the stack size for subthreads has been reconfigured in the
   // ArangoGlobalContext above in the libmusl case:
   int result;
-  auto runTests = [&result] (int argc, char* argv[]) {
-      result = Catch::Session().run( argc, argv );
+  auto tests = [] (int argc, char* argv[]) -> int {
+    return Catch::Session().run( argc, argv );
   };
-  std::thread subthread(runTests, argc, argv);
-  subthread.join();
+  TestThread<decltype(tests)> t(std::move(tests), argc, argv);
+  result = t.result();
```

Comment:
Our of curiousity don't we need the `join()` some where?
I do not see any place waiting for the test to complete. On the old we did wait for the subthread.

Reply:
No don't need join. The thread is joined or detatched in destruction of arangodb::Thread.